drivers/w1/w1_ds248x.c

```diff
@@ -0,0 +1,200 @@
+{
+	int ret;
+
+	const uint8_t buf[] = {cmd, data ? *data : 0};
+
+	ret = i2c_write_dt(spec, buf, data ? 2 : 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int ds248x_read(const struct i2c_dt_spec *spec, uint8_t rp, uint8_t *reg)
+{
+	int ret;
+
+	switch (rp) {
+	case REG_NONE:
+		/*
+		 * Special value: Don't change read pointer
+		 */
+		break;
+	case REG_PORT:
+	case REG_CONFIG:
+	case REG_CHANNEL:
+	case REG_DATA:
+	case REG_STATUS:
+		ret = ds248x_write(spec, CMD_SRP, &rp);
+		if (ret < 0) {
+			return ret;
+		}
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	ret = i2c_read_dt(spec, reg, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+int ds248x_reset_bus(const struct i2c_dt_spec *spec)
+{
+	int ret;
+
+	uint8_t reg;
+
+	ret = ds248x_write(spec, CMD_1WRS, NULL);
+	if (ret < 0) {
+		return ret;
+	}
+
+	do {
+		ret = ds248x_read(spec, REG_NONE, &reg);
+		if (ret < 0) {
+			return ret;
+		}
+	} while (reg & STATUS_1WB_msk);
+
+	return reg & STATUS_PPD_msk ? 1 : 0;
+}
+
+int ds248x_reset_device(const struct i2c_dt_spec *spec)
+{
+	int ret;
+
+	uint8_t reg;
+
+	ret = ds248x_write(spec, CMD_DRST, NULL);
+	if (ret < 0) {
+		return ret;
+	}
+
+	do {
+		ret = ds248x_read(spec, REG_NONE, &reg);
+		if (ret < 0) {
+			return ret;
+		}
+	} while (!(reg & STATUS_RST_msk));
+
+	return 0;
+}
+
+static int ds248x_single_bit(const struct i2c_dt_spec *spec, uint8_t bit_msk)
+{
+	int ret;
+
+	uint8_t reg;
+
+	ret = ds248x_write(spec, CMD_1WSB, &bit_msk);
+	if (ret < 0) {
+		return ret;
+	}
+
+	do {
+		ret = ds248x_read(spec, REG_NONE, &reg);
+		if (ret < 0) {
+			return ret;
+		}
+	} while (reg & STATUS_1WB_msk);
+
+	return reg & STATUS_SBR_msk;
+}
+
+int ds248x_read_bit(const struct i2c_dt_spec *spec)
+{
+	int ret = ds248x_single_bit(spec, BIT_SET_msk);
+
+	return ret > 1 ? 1 : ret;
+}
+
+int ds248x_write_bit(const struct i2c_dt_spec *spec, bool bit)
+{
+	int ret = ds248x_single_bit(spec, bit ? BIT_SET_msk : BIT_CLR_msk);
+
+	return ret > 0 ? 0 : ret;
+}
+
+int ds248x_read_byte(const struct i2c_dt_spec *spec)
+{
+	int ret;
+
+	uint8_t reg;
+
+	ret = ds248x_write(spec, CMD_1WRB, NULL);
+	if (ret < 0) {
+		return ret;
+	}
+
+	do {
+		ret = ds248x_read(spec, REG_NONE, &reg);
+		if (ret < 0) {
+			return ret;
+		}
+	} while (reg & STATUS_1WB_msk);
+
+	ret = ds248x_read(spec, REG_DATA, &reg);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return reg;
+}
+
+int ds248x_write_byte(const struct i2c_dt_spec *spec, uint8_t byte)
+{
+	int ret;
+
+	uint8_t reg;
+
+	ret = ds248x_write(spec, CMD_1WWB, &byte);
+	if (ret < 0) {
+		return ret;
+	}
+
+	do {
+		ret = ds248x_read(spec, REG_NONE, &reg);
+		if (ret < 0) {
+			return ret;
+		}
+	} while (reg & STATUS_1WB_msk);
+
+	return 0;
+}
+
+int ds248x_write_config(const struct i2c_dt_spec *spec, uint8_t cfg)
+{
+	int ret;
+
+	if (cfg & ~(DEVICE_APU_msk | DEVICE_PDN_msk | DEVICE_SPU_msk | DEVICE_1WS_msk)) {
+		return -EINVAL;
+	}
+
+	uint8_t reg = cfg | ~cfg << 4;
```

Comment:
move the definition to the beginning of the function.

Reply:
This assignment is (technically) not always necessary; therefore the early exit was above. Anyway doesn't really make a difference. Done.